### PR TITLE
Prevents deepfrying space

### LINKED ISSUE #40527: Fried space
[Round ID]: # 94501
situation in the picture
dreamseeker_2018-09-26_16-24-05

### PR DESCRIPTION
**Pull request by MrDoomBringer**
Fixes #40527
:cl: MrDoomBringer
fix: massive service department nerf: space can no longer be extra crispy.
/:cl:

### CODE REVIEW

**Comment by sagattyy** [comment]
this removes fun

**Comment by Potato-Masher** [comment]
What the fuck

Lmao I love how ridiculous this is.

**Comment by F-OS** [comment]
improve don't remove

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 1]
```diff
@@ -136,7 +136,7 @@
 	return TRUE
 
 /datum/reagent/consumable/cooking_oil/reaction_turf(turf/open/T, reac_volume)
-	if(!istype(T))
+	if(!istype(T) || istype(T, /turf/open/space))
```
Comment (ShizCalev) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: isspaceturf(T)

should probably also check lava & chasms too.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 1 above)
Comment (MrDoomBringer) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: @ShizCalev do we have a list of turfs like these

**Comment by FrostyFridge** [comment]
>Tag: Fix

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 1 above)
Comment (MrDoomBringer) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: i added one

**Comment by ShizCalev** [comment]
@davethwave reaction_turf() will fix it for everything.

**Comment by davethwave** [comment]
Hmm, the best way to deep fry space I believe is the use of frying oil grenades. Though I haven't tested it it should work hopefully.

File: code/game/objects/effects/decals/decal.dm [hunk 2]
```diff
@@ -15,7 +15,7 @@
 		qdel(src)
 
 /obj/effect/decal/proc/NeverShouldHaveComeHere(turf/T)
-	return isspaceturf(T) || isclosedturf(T) || islava(T) || istype(T, /turf/open/water) || ischasm(T)
+	return isclosedturf(T) || isgroundlessturf(T)
```
Comment (ShizCalev) [inline code/game/objects/effects/decals/decal.dm]: )

Comment (MrDoomBringer) [reply code/game/objects/effects/decals/decal.dm]: wait what where do i put the close parenthesis

Comment (Qustinnus) [reply code/game/objects/effects/decals/decal.dm]: nowhere.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 3]
```diff
@@ -136,7 +136,7 @@
 	return TRUE
 
 /datum/reagent/consumable/cooking_oil/reaction_turf(turf/open/T, reac_volume)
-	if(!istype(T))
+	if(!istype(T) || isgroundlessturf(T)))
```
Comment (Qustinnus) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: remove one here you doofus

**Comment by MrDoomBringer** [comment]
Wtf goof hacked my code and put that there I swear

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 3 above)
Comment (MrDoomBringer) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Wtf goof hacked my code and put that there I swear

**Review by SpaceManiac — DISMISSED** [review]

**Comment by SpaceManiac** [comment]
approved except for the part where it doesn't compile

**Comment by MrDoomBringer** [comment]
writing code that compiles is a sign of weakness change my mind

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**